### DISCUSSION TITLE
Pyodide npm package and micropip wheel

### DISCUSSION THREAD

**Discussion by sgpinkus** [pyodide/pyodide#5155] [Q&A] [answered]
I have a vue+vite  app. I was just downloading the pyodide.js from CDN and it was working fine. I'm trying to switch to the npm package but I'm encountering issues in dev and prod builds. In dev the pyodide.asm.js is trying to fetch `http://localhost:5173/node_modules/pyodide/micropip-0.6.0-py3-none-any.whl` and `http://localhost:5173/node_modules/pyodide/packaging-23.2-py3-none-any.whl`. These files don't exist in the npm package, and even if they did I don't think the vite dev server would let you retrieve them directly from node_modules. The dev server is just catching this request and returning the index.html file resulting in integrity check issues:

> None of the “sha256” hashes in the integrity attribute match the content of the subresource. The computed hash is “Hf7TwzJz+eoDX67OpNlBK2iNz1JWfLxEpYTiWsK7R1w=”.

I'm confused about how this is supposed to work. The only doc I could find on it is here -- https://pyodide.org/en/stable/usage/working-with-bundlers.html. I've used that vite config. But it doesn't mention anything about how pyodide.js will try and fetch the micropip wheel. Please help.

**Comment by sgpinkus** [comment, marked as answer]
Oh oops, didn't RTFD properly - you need to download an archive of wheels and stick it somewhere accessible to your app then tell `loadPyodide()` where to find it. 

> Download and extract Pyodide packages from [Github releases](https://github.com/pyodide/pyodide/releases) (pyodide-build-*.tar.bz2). The version of the release needs to match exactly the version of this package. -- https://www.npmjs.com/package/pyodide

**Reply by hoodmane** [reply]
That information is a bit outdated, we should update it. In a normal node application, `pyodide.loadPackage()` will automatically download the package from jsdelivr and cache it into `node_modules/pyodide`. I think vite needs a plugin to help it understand this. The most recent instructions for working with vite are here:
https://pyodide.org/en/stable/usage/working-with-bundlers.html#vite
